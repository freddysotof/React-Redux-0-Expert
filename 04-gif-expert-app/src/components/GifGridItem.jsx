import PropTypes from 'prop-types'
export const GifGridItem = ({title,url}) => {
  return (
    <div className="card">
        <img src={url} alt={title}/>
        <p>{title}</p>
    </div>
    // <li></li> 
  )

}

GifGridItem.propTypes={
  title:PropTypes.string.isRequired,
  url:PropTypes.string.isRequired
}



