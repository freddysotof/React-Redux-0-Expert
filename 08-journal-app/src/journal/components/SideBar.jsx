import { TurnedInNot } from "@mui/icons-material"
import { Box, Divider, Drawer, Grid, List, ListItem, ListItemButton, ListItemIcon, ListItemText, Toolbar, Typography } from "@mui/material"

export const SideBar = ({ drawerWidth = 240 }) => {
    return (
        <Box
            component='nav'
            sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}
        >
            <Drawer
                variant="permanent" // temporary
                open
                // onClose={}
                sx={{
                    display: { xs: 'block' },
                    '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth }
                }}
            >
                <Toolbar>
                    <Typography variant="h6" noWrap component="div">
                        Freddy Soto
                    </Typography>


                </Toolbar>
                <Divider />
                <List>
                    {
                        ['Enero', 'Febrero', 'Marzo', 'Abril'].map(text => (
                            <ListItem key={text}>
                                <ListItemButton>
                                    <ListItemIcon>
                                        <TurnedInNot />
                                    </ListItemIcon>
                                    <Grid container>
                                        <ListItemText primary={text} />
                                        <ListItemText secondary={'Do laborum et magna sit ipsum ad ex amet laboris aliqua incididunt proident irure.'} />
                                    </Grid>
                                </ListItemButton>
                            </ListItem>
                        ))
                    }
                </List>
            </Drawer>
        </Box>
    )
}
