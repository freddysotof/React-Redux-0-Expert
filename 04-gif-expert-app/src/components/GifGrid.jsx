import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

const {images, isLoading}= useFetchGifs(category);
// console.log(isLoading)
  
    
    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && ( <h2>Cargando...</h2> )
            }
            <div className="card-grid">
                {/* <LoadingMessage isLoading={isLoading} */}
                {
                
                    images.map(image => (
                        <GifGridItem 
                        key={image.id} 
                        {...image}
                        />

                    ))
                    
                }
            </div>
        </>
    )
}

GifGrid.propTypes={
    category:PropTypes.string.isRequired,
}
