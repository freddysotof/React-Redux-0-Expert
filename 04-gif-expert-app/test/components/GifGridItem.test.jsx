import { render,screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components/GifGridItem"

describe('Pruebas en <GifGridItem />',()=>{
    const title='Saitama'
    const url='https://google.com/';


    test('debe hacer match con snapshot', () => { 

        const {container}=    render(<GifGridItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
     })

     test('debe de mostrar la image con el url y el alt indicado', () => { 
        
        render(<GifGridItem title={title} url={url} />);
        // screen.debug();
        //expect(screen.getByRole('img').src).toBe(url);
        const {src,alt }=screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
      })

      test('debe de mostrar el titulo en el componente', () => {
        render(<GifGridItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();

       })
})