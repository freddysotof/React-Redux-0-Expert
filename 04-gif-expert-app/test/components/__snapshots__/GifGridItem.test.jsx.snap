// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://google.com/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
